### PR TITLE
Add support for `PodIdentityAssociation.eks.aws.upbound.io/v1beta1` resource

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

This PR adds support for `PodIdentityAssociation.eks.aws.upbound.io/v1beta1` resource.

Fixes https://github.com/upbound/provider-aws/issues/985

I have:

- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Uptest run: https://github.com/upbound/provider-aws/actions/runs/7739049201

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/eks/v1beta1/podidentityassociation.yaml"

File: config/externalname.go [hunk 1]
```diff
@@ -42,6 +42,11 @@ var TerraformPluginFrameworkExternalNameConfigs = map[string]config.ExternalName
 	// AppConfig Environments can be imported by using the environment ID and application ID separated by a colon (:)
 	// terraform-plugin-framework
 	"aws_appconfig_environment": appConfigEnvironment(),
+
+	// eks
+	//
+	// Imported using the cluster_name and association_id separated by a comma (,)
```
Comment (erhancagirici) [inline config/externalname.go:48]: This import syntax does not apply to the framework architecture, to import consumers should set `association_id` as the external name and pass `cluster_name` in `spec.forProvider`

**Review by erhancagirici — CHANGES_REQUESTED** [review]
Left a comment regarding the import syntax doc comment, otherwise LGTM 👍

File: config/externalname.go (hunk 1 above)
Comment (turkenf) [reply config/externalname.go]: Thanks for the catch, done.

**Pull request merged**